src/main/java/wiresegal/wob/arcanum/Parsing.kt

```diff
@@ -167,6 +173,78 @@ fun about(message: Message) {
     })
 }
 
+class Progress(title: String, percentage: String, link: String?) {
+    operator fun component1(): String {
+        return title
+    }
+    operator fun component2(): String {
+        return percentage
+    }
+    operator fun component3(): String? {
+        return link
+    }
+
+    private val title: String = title
+    private val percentage: String = percentage
+    private val link: String? = link
+}
+
+val progressCache = mutableListOf<Progress>()
+var progressCacheTimeStamp: Long? = null
+
+fun showProgressBar(message: Message) {
+    val progresses = extractProgresses()
+    val embed = EmbedBuilder ()
+            .setColor(embedColor)
+            .setTitle("Progress Bars")
+            .setUrl(homepageTarget)
+
+    val full = "█"
+    val empty = "░"
+
+    for (progress in progresses) {
+        val (name, percent, link) = progress
+        val percentNumber: Int = Integer.parseInt(percent)
+        val sensitivity = 20
+        val points: Int = (percentNumber/100.0 * sensitivity).toInt()
+        val bar = full.repeat(points) + empty.repeat(sensitivity - points)
+        val linkText = if(link != null) "_[check it out]($link)_\n" else ""
+        embed.addField("**$name**", "$linkText$bar $percent%", false).setUrl("https://google.com/")
+    }
+    message.channel.sendMessage(embed)
+}
+
+fun extractProgresses(): MutableList<Progress> {
+    if (progressCacheTimeStamp != null
+            && (System.currentTimeMillis() - progressCacheTimeStamp!! <= progressCachePersistence))
+        return progressCache
+
+    val html = URL(homepageTarget).readText()
+    val parentClass = "vc_label"
+
+    progressCache.clear()
+
+    val s: Document? = Jsoup.parse(html);
```

Comment:
This variable could use a more descriptive name and I don't see why it should be nullable. Even if Jsoup returns null, you force it to be non-null in the loop. You should be able to remove the type ascription. Same for the `content` val.